File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-13.js
class Solution {
    /**
     * @param {string} s
     * @return {number}
     */
    lengthOfLongestSubstring(s) {
        let set = new Set();
        let l = 0;
        let res = 0;
        for (let r = 0; r < s.length; r ++) {
            let cur = s[r];
            while(set.has(cur)) {
                set.delete(s[l]);
                l ++;
            }

            set.add(cur);
            let curLen = r - l + 1;
            res = Math.max(res, curLen)
        }

        return res;
    }
}

File: Data Structures & Algorithms/daily-temperatures/submission-13.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        res = [ 0 for _ in range(len(temperatures)) ] 

        for i in range(len(temperatures)): 
            cur = temperatures[i] 

            for j in range(i + 1, len(temperatures)): 
                nxt = temperatures[j] 

                if nxt > cur: 
                    res[i] = j - i 
                    break 

        return res